In [8]:
import pandas as pd

In [9]:
variable_dict_df = pd.read_csv('Data/American Community Survey Data/acs5_variable_dict_2014_2019.csv')

In [10]:
variable_dict_df.head()

variable_code                                              label  \
0   DP02_0019EA  Annotation of Estimate!!RELATIONSHIP!!Populati...   
1    DP02_0126E       Estimate!!ANCESTRY!!Total population!!Danish   
2   DP02_0072EA  Annotation of Estimate!!DISABILITY STATUS OF T...   
3  DP02_0069PMA  Annotation of Percent Margin of Error!!VETERAN...   
4    DP02_0126M  Margin of Error!!ANCESTRY!!Total population!!D...   

                                             concept predicateType group  \
0  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...        string  DP02   
1  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...           int  DP02   
2  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...        string  DP02   
3  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...        string  DP02   
4  SELECTED SOCIAL CHARACTERISTICS IN THE UNITED ...           int  DP02   

   limit  predicateOnly  acs_year  
0      0           True      2014  
1      0           True      2014  
2      0           True      2014  
3      0           True      2014  
4      0           True      2014

In [11]:
variable_dict_df.dtypes

variable_code    object
label            object
concept          object
predicateType    object
group            object
limit             int64
predicateOnly      bool
acs_year          int64
dtype: object

In [12]:
#Count the No. of variable codes in 2014 and 2019
variable_dict_df.groupby('acs_year')['variable_code'].count()

acs_year
2014    4088
2019    4176
Name: variable_code, dtype: int64

There are more variable codes in 2019 than in 2014. To be exact, there are 88 extra codes in 2019

In [13]:
# Check what variable_codes appear in 2014 set alone or 2019 set alone
variable_dict_2014 = variable_dict_df.loc[variable_dict_df['acs_year']==2014]
variable_dict_2019 = variable_dict_df.loc[variable_dict_df['acs_year']==2019]

In [14]:
# Find variable codes that appear in 2014 but not in 2019
variable_code_2019 = variable_dict_2019['variable_code'].tolist()
for variable_code in variable_dict_2014['variable_code']:
    if variable_code not in variable_code_2019:
        print(variable_code)

All varible codes that appear in 2014 also appear in 2019

In [15]:
# Find variable codes that appear in 2019 but not in 2014
variable_code_2014 = variable_dict_2014['variable_code'].tolist()
for variable_code in variable_dict_2019['variable_code']:
    if variable_code not in variable_code_2014:
        print(variable_code)

DP02_0153PEA
DP02_0153PMA
DP02_0153EA
DP02_0153MA
DP02_0153PE
DP02_0153PM
DP02_0153M
DP02_0153E
DP04_0143PEA
DP04_0143PMA
DP04_0142EA
DP04_0142MA
DP04_0142PM
DP04_0142PE
DP04_0143EA
DP04_0143MA
DP04_0143PM
DP04_0143PE
DP04_0142PEA
DP04_0142PMA
DP04_0143M
DP04_0143E
DP04_0142M
DP04_0142E
DP05_0089PMA
DP05_0089PEA
DP05_0082EA
DP05_0082MA
DP05_0082PM
DP05_0082PE
DP05_0083EA
DP05_0084PMA
DP05_0084PEA
DP05_0088EA
DP05_0088M
DP05_0088E
DP05_0087M
DP05_0089M
DP05_0089E
DP05_0087PMA
DP05_0084E
DP05_0083M
DP05_0085E
DP05_0084M
DP05_0088MA
DP05_0086E
DP05_0088PM
DP05_0085M
DP05_0088PE
DP05_0087E
DP05_0086M
DP05_0087PEA
DP05_0082E
DP05_0083E
DP05_0082M
DP05_0089EA
DP05_0089MA
DP05_0089PM
DP05_0089PE
DP05_0085PM
DP05_0085PE
DP05_0082PEA
DP05_0082PMA
DP05_0085PMA
DP05_0086EA
DP05_0085PEA
DP05_0086MA
DP05_0086PM
DP05_0086PE
DP05_0088PMA
DP05_0088PEA
DP05_0087EA
DP05_0087MA
DP05_0087PM
DP05_0087PE
DP05_0083MA
DP05_0083PM
DP05_0083PE
DP05_0083PMA
DP05_0084EA
DP05_0084MA
DP05_0083PEA
DP05_0084PM
DP05_0

In [18]:
#count the variable codes whose labels are different between 2014 and 2019
count_mismatched_labels = 0        
for var_code in variable_code_2014:
    label_2014_series = variable_dict_2014.loc[variable_dict_2014['variable_code']==var_code]['label']
    label_2014 = label_2014_series[label_2014_series.index[0]] #since only one entry is available for each variable code in agiven year
    label_2019_series = variable_dict_2019.loc[variable_dict_2019['variable_code']==var_code]['label']
    label_2019 = label_2019_series[label_2019_series.index[0]] #since only one entry is available for each variable code in agiven year
    if  label_2014 !=  label_2019:
        count_mismatched_labels +=1
        print(var_code)

DP02_0019EA
DP02_0126E
DP02_0072EA
DP02_0069PMA
DP02_0126M
DP02_0057PMA
DP02_0125M
DP02_0054PM
DP02_0041MA
DP02_0070PMA
DP02_0125E
DP02_0054PE
DP02_0108MA
DP02_0124M
DP02_0124E
DP02_0098MA
DP02_0082PMA
DP02_0006EA
DP02_0139EA
DP02_0045PMA
DP02_0123M
DP02_0123E
DP02_0122M
DP02_0082PEA
DP02_0122E
DP02_0143PE
DP02_0130MA
DP02_0143PM
DP02_0121M
DP02_0085EA
DP02_0010PE
DP02_0121E
DP02_0010PM
DP02_0130PE
DP02_0130PM
DP02_0120M
DP02_0067PM
DP02_0120E
DP02_0054MA
DP02_0067PE
DP02_0094PEA
DP02_0123PMA
DP02_0085PM
DP02_0132PEA
DP02_0072MA
DP02_0085PE
DP02_0019MA
DP02_0113EA
DP02_0005PEA
DP02_0144PEA
DP02_0037EA
DP02_0100EA
DP02_0111PMA
DP02_0090EA
DP02_0019PM
DP02_0139MA
DP02_0017PEA
DP02_0072PM
DP02_0072PE
DP02_0019PE
DP02_0006MA
DP02_0094PMA
DP02_0041PE
DP02_0126EA
DP02_0041PM
DP02_0042PEA
DP02_0129E
DP02_0098PM
DP02_0129M
DP02_0085MA
DP02_0098PE
DP02_0128E
DP02_0128M
DP02_0029PEA
DP02_0127E
DP02_0108PE
DP02_0030PEA
DP02_0108PM
DP02_0127M
DP02_0123PEA
DP02_0084M
DP02_0085M
DP02_0132PMA
DP02_00

DP02_0122PE
DP02_0122PM
DP02_0044M
DP02_0064EA
DP02_0044E
DP02_0046PM
DP02_0043M
DP02_0033MA
DP02_0043E
DP02_0046PE
DP02_0042M
DP02_0042E
DP02_0122MA
DP02_0136PMA
DP02_0040M
DP02_0135PE
DP02_0059MA
DP02_0041M
DP02_0135PM
DP02_0065PMA
DP02_0041E
DP02_0118PEA
DP02_0040E
DP02_0020EA
DP02_0002PM
DP02_0140EA
DP02_0148PMA
DP02_0077EA
DP02_0077PMA
DP02_0002PE
DP02_0013PEA
DP02_0059PM
DP02_0046MA
DP02_0059PE
DP02_0106PEA
DP02_0029E
DP02_0020PE
DP02_0095EA
DP02_0046PMA
DP02_0105EA
DP02_0029M
DP02_0020PM
DP02_0028E
DP02_0095PEA
DP02_0077PM
DP02_0028M
DP02_0064MA
DP02_0077PE
DP02_0027E
DP02_0027M
DP02_0029EA
DP02_0140PE
DP02_0034PMA
DP02_0140PM
DP02_0026M
DP02_0026E
DP02_0082EA
DP02_0051EA
DP02_0025M
DP02_0025E
DP02_0124PMA
DP02_0071PEA
DP02_0033PE
DP02_0020MA
DP02_0131PEA
DP02_0024M
DP02_0033PM
DP02_0024E
DP02_0058PMA
DP02_0140MA
DP02_0023M
DP02_0077MA
DP02_0023E
DP02_0143PEA
DP02_0022M
DP02_0118EA
DP02_0022E
DP02_0083PEA
DP02_0112PMA
DP02_0118PE
DP02_0100PMA
DP02_0105MA
DP02_0118PM
DP02_0095MA


DP03_0057PMA
DP03_0073E
DP03_0126PE
DP03_0131PEA
DP03_0072M
DP03_0126PM
DP03_0068EA
DP03_0082E
DP03_0081M
DP03_0059PM
DP03_0083E
DP03_0059PE
DP03_0082M
DP03_0080E
DP03_0077EA
DP03_0053PEA
DP03_0081E
DP03_0080M
DP03_0122MA
DP03_0111PEA
DP03_0135PE
DP03_0016PMA
DP03_0051EA
DP03_0135PM
DP03_0123PEA
DP03_0122PE
DP03_0122PM
DP03_0064EA
DP03_0105EA
DP03_0090PEA
DP03_0129M
DP03_0077MA
DP03_0129E
DP03_0053PMA
DP03_0135PEA
DP03_0089PEA
DP03_0111PMA
DP03_0127M
DP03_0077PEA
DP03_0064PM
DP03_0051MA
DP03_0064PE
DP03_0127E
DP03_0118EA
DP03_0123PMA
DP03_0082EA
DP03_0125E
DP03_0125M
DP03_0065PEA
DP03_0126E
DP03_0126M
DP03_0077PM
DP03_0064MA
DP03_0077PE
DP03_0135E
DP03_0118PE
DP03_0135M
DP03_0093PEA
DP03_0105MA
DP03_0118PM
DP03_0136E
DP03_0136M
DP03_0051PM
DP03_0051PE
DP03_0056PMA
DP03_0016EA
DP03_0133E
DP03_0123EA
DP03_0133M
DP03_0134E
DP03_0134M
DP03_0105PE
DP03_0105PM
DP03_0081PEA
DP03_0135PMA
DP03_0131E
DP03_0131M
DP03_0132E
DP03_0118MA
DP03_0132M
DP03_0082MA
DP03_0130E
DP03_0130M
DP03_0136EA
DP03_

DP04_0005PE
DP04_0091PEA
DP04_0042PMA
DP04_0049MA
DP04_0125MA
DP04_0110M
DP04_0138PE
DP04_0110E
DP04_0138PM
DP04_0029PMA
DP04_0030PMA
DP04_0023EA
DP04_0066PMA
DP04_0112PE
DP04_0067M
DP04_0132PMA
DP04_0112PM
DP04_0049PM
DP04_0017PMA
DP04_0067E
DP04_0049PE
DP04_0036MA
DP04_0005PMA
DP04_0066M
DP04_0078PMA
DP04_0066E
DP04_0130EA
DP04_0069M
DP04_0054PEA
DP04_0069E
DP04_0112MA
DP04_0068M
DP04_0068E
DP04_0125PE
DP04_0125PM
DP04_0067EA
DP04_0116M
DP04_0116E
DP04_0115M
DP04_0115E
DP04_0104PEA
DP04_0041EA
DP04_0118M
DP04_0054EA
DP04_0128PEA
DP04_0118E
DP04_0117M
DP04_0066PEA
DP04_0036PM
DP04_0117E
DP04_0023MA
DP04_0036PE
DP04_0112M
DP04_0112E
DP04_0078PEA
DP04_0111M
DP04_0130MA
DP04_0111E
DP04_0085EA
DP04_0114M
DP04_0114E
DP04_0054PMA
DP04_0108EA
DP04_0023PM
DP04_0098EA
DP04_0023PE
DP04_0113M
DP04_0113E
DP04_0116PEA
DP04_0067MA
DP04_0094PMA
DP04_0019EA
DP04_0072EA
DP04_0054PM
DP04_0104PMA
DP04_0054PE
DP04_0041MA
DP04_0130PE
DP04_0128PMA
DP04_0130PM
DP04_0067PM
DP04_0054MA
DP04_0067PE
DP04_0108M


DP04_0029EA
DP04_0082EA
DP04_0099M
DP04_0118PE
DP04_0099E
DP04_0105MA
DP04_0118PM
DP04_0098M
DP04_0136EA
DP04_0098E
DP04_0071PEA
DP04_0097M
DP04_0083PEA
DP04_0095MA
DP04_0118MA
DP04_0051PM
DP04_0018PEA
DP04_0051PE
DP04_0095E
DP04_0095PEA
DP04_0094M
DP04_0094E
DP04_0082PE
DP04_0105PE
DP04_0047EA
DP04_0105PM
DP04_0100PEA
DP04_0093M
DP04_0097E
DP04_0082MA
DP04_0123EA
DP04_0096M
DP04_0096E
DP04_0029MA
DP04_0095PM
DP04_0095M
DP04_0095PE
DP05_0050PE
DP05_0025PEA
DP05_0015EA
DP05_0050PM
DP05_0037PEA
DP05_0081MA
DP05_0059EA
DP05_0049PEA
DP05_0013PMA
DP05_0028MA
DP05_0050PEA
DP05_0028M
DP05_0077PMA
DP05_0028PM
DP05_0028E
DP05_0028PE
DP05_0081PM
DP05_0015MA
DP05_0081PE
DP05_0029M
DP05_0029E
DP05_0046EA
DP05_0033EA
DP05_0065PMA
DP05_0024M
DP05_0024E
DP05_0059MA
DP05_0025M
DP05_0025E
DP05_0015PM
DP05_0015PE
DP05_0026M
DP05_0026E
DP05_0077EA
DP05_0013PEA
DP05_0027M
DP05_0027E
DP05_0020EA
DP05_0077PEA
DP05_0020E
DP05_0028PMA
DP05_0021E
DP05_0059PM
DP05_0059PE
DP05_0046MA
DP05_0020M
DP05_0022M
DP05_0

In [19]:
print('No of variable codes whose labels are different in 2014 and 2019 : {}'.format(count_mismatched_labels))

No of variable codes whose labels are different in 2014 and 2019 : 3544


Out of the 4088 common variable codes between 2014 and 2019, 86.69% of the variable codes didn't have the same label for both years. This is however based on an exact string comparison.